stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/compression/UnzipBytes.java

```diff
@@ -0,0 +1,177 @@
+/*
+ *  Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.ballerinalang.nativeimpl.compression;
+
+import org.ballerinalang.bre.Context;
+import org.ballerinalang.bre.bvm.BlockingNativeCallableUnit;
+import org.ballerinalang.model.types.TypeKind;
+import org.ballerinalang.natives.annotations.Argument;
+import org.ballerinalang.natives.annotations.BallerinaFunction;
+import org.ballerinalang.util.exceptions.BLangRuntimeException;
+
+import java.io.BufferedOutputStream;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+/**
+ * Native function ballerina.compression:unzipBytes.
+ *
+ * @since 0.970.0
+ */
+@BallerinaFunction(
+        orgName = "ballerina", packageName = "compression",
+        functionName = "unzipBytes",
+        args = {@Argument(name = "content", type = TypeKind.BLOB),
+                @Argument(name = "destDir", type = TypeKind.STRING),
+                @Argument(name = "folderToUnzip", type = TypeKind.STRING)},
+        isPublic = true
+)
+public class UnzipBytes extends BlockingNativeCallableUnit {
+    /**
+     * File content as byte array defined.
+     */
+    private static final int SRC_AS_BYTEARRAY_FIELD_INDEX = 0;
+    /**
+     * File path of the destination directory.
+     */
+    private static final int DEST_PATH_FIELD_INDEX = 0;
+
+    /**
+     * Folder to unzip from the compressed bytes.
+     */
+    private static final int FOLDER_TO_UNZIP_INDEX = 1;
+
+    /**
+     * Decompress/unzip byte arrays/blob.
+     *  @param fileContentAsByteArray file content as a byte arry
+     * @param outputFolder           destination folder
+     * @param folderToUnzip folder to unzip in the zipped file
+     */
+    static void decompress(byte[] fileContentAsByteArray, String outputFolder, String folderToUnzip) {
```

Comment:
Shall change this to `static void decompress(InputStream zippedBytes, String outputFolder, String folderToUnzip)`

Then you will be able to unzip a file without reading it fully first.

Reply:
This method takes in a byte array of a compressed file and then unzips it.

Reply:
Fixed with bc105af